### DISCUSSION TITLE
[Q] Is there way to add vba code to xls file?

### DISCUSSION THREAD

**Discussion by domadev812** [exceljs/exceljs#2435] [Q&A]
## 💬 Questions and Help
Hi, ExcelJS dev team.

I have been building angular app using ExcelJS and I face the one problem and I'd like you to help.
The purpose of app is to download excel template file so that user can input new data easily.
After user finish entering data, they should upload file to save data into database.
By the way I just need to add some vba code into template file for validation input data.
The template file is generated dynamically in app.

I just want to know there is way to add vba code into download xls file.

Thanks,

Istvan

**Comment by GowthamGottimukkala** [comment]
I am working on a project that exactly need this. Did you solve this?

**Comment by domadev812** [comment]
I tried a lot of times, but there is no solution.  :(

**Comment by adnaanbheda** [comment]
Hey, I'm currently pursuing this and found an unusual solution.

So, I still don't know if VBA injection works in this library, or even if it's copied when opening a file and writing it somewhere.
SheetJS supports reading/writing an existing VBA blob, so what you can do is create a template file that has your VBA code / data validation and add data in other sheets and use those sheets as your data source to generate dropdown lists and what not. You can hide those sheets so the user can't access them when you create that template file.

So, you read that file and the VBA code comes along with it, edit that file with your new data and send this file back to the user.
